decimal_test.go

```diff
@@ -743,3 +743,101 @@ func TestSizeof(t *testing.T) {
 		t.Errorf("sizeof(Context) changed: %d", s)
 	}
 }
+
+func TestUnmarshalJSON(t *testing.T) {
```

Comment:
Let's copy the test structure from math/big too: https://github.com/golang/go/blob/master/src/math/big/intmarsh_test.go#L77

Except use CmpTotal instead of Cmp.